CMakeLists.txt

```diff
@@ -0,0 +1,630 @@
+    src/loaders/prowizard/np2.c
+    src/loaders/prowizard/np3.c
+    src/loaders/prowizard/p40.c
+    src/loaders/prowizard/p61a.c
+    src/loaders/prowizard/pha.c
+    src/loaders/prowizard/pm10c.c
+    src/loaders/prowizard/pm18a.c
+    src/loaders/prowizard/pp10.c
+    src/loaders/prowizard/pp21.c
+    src/loaders/prowizard/prowiz.c
+    src/loaders/prowizard/prun1.c
+    src/loaders/prowizard/prun2.c
+    src/loaders/prowizard/ptktable.c
+    src/loaders/prowizard/skyt.c
+    src/loaders/prowizard/starpack.c
+    src/loaders/prowizard/tdd.c
+    src/loaders/prowizard/theplayer.c
+    src/loaders/prowizard/titanics.c
+    src/loaders/prowizard/tp1.c
+    src/loaders/prowizard/tp3.c
+    src/loaders/prowizard/tuning.c
+    src/loaders/prowizard/unic2.c
+    src/loaders/prowizard/unic.c
+    src/loaders/prowizard/wn.c
+    src/loaders/prowizard/xann.c
+    src/loaders/prowizard/zen.c
+)
+
+set(LIBXMP_SRC_LIST_DEPACKERS
+    src/depackers/arcfs.c
+    src/depackers/bunzip2.c
+    src/depackers/crc32.c
+    src/depackers/depacker.c
+    src/depackers/gunzip.c
+    src/depackers/inflate.c
+    src/depackers/mmcmp.c
+    src/depackers/muse.c
+    src/depackers/oxm.c
+    src/depackers/ppdepack.c
+    src/depackers/readlzw.c
+    src/depackers/readrle.c
+    src/depackers/s404_dec.c
+    src/depackers/unarc.c
+    src/depackers/uncompress.c
+    src/depackers/unlha.c
+    src/depackers/unlzx.c
+    src/depackers/unsqsh.c
+    src/depackers/unxz.c
+    src/depackers/unzip.c
+    src/depackers/vorbis.c
+    src/depackers/xfnmatch.c
+    src/depackers/xz_dec_lzma2.c
+    src/depackers/xz_dec_stream.c
+)
+
+if(AMIGA)
+    list(APPEND LIBXMP_SRC_LIST_DEPACKERS
+        src/depackers/xfd.c
+        src/depackers/xfd_link.c
+    )
+endif()
+
+if(WIN32)
+    list(APPEND LIBXMP_SRC_COMMON
+        src/win32.c
+        src/win32/ptpopen.c
+    )
+    if(MSVC AND CMAKE_BUILD_TYPE_LOWER STREQUAL "debug")
+        list(APPEND LIBXMP_SRC_COMMON
+            src/win32/debug.c
+        )
+    endif()
+endif()
+
+set(LIBXMP_DEFINES)
+
+if(DISABLE_IT)
+    list(APPEND LIBXMP_DEFINES -DLIBXMP_CORE_DISABLE_IT)
+endif()
+
+if(LIBXMP_DISABLE_DEPACKERS)
+    list(APPEND LIBXMP_DEFINES -DLIBXMP_NO_DEPACKERS)
+else()
+    list(APPEND LIBXMP_SRC_LIST_FULL_ONLY ${LIBXMP_SRC_LIST_DEPACKERS})
+endif()
+
+if(LIBXMP_DISABLE_PROWIZARD)
+    list(APPEND LIBXMP_DEFINES -DLIBXMP_NO_PROWIZARD)
+else()
+    list(APPEND LIBXMP_SRC_LIST_FULL_ONLY ${LIBXMP_SRC_LIST_PROWIZARD})
+endif()
+
+list(APPEND LIBXMP_SRC_LITE ${LIBXMP_SRC_LIST_LITE} ${LIBXMP_SRC_LIST_LITE_ONLY} ${LIBXMP_SRC_COMMON})
+list(APPEND LIBXMP_SRC_FULL ${LIBXMP_SRC_LIST_LITE} ${LIBXMP_SRC_LIST_FULL_ONLY} ${LIBXMP_SRC_LIST_FULL} ${LIBXMP_SRC_COMMON})
+
+set(XMP_INSTALLS)
+
+# ============= FULL =============
+if(LIBXMP_BUILD_FULL AND BUILD_STATIC)
+    add_library(xmp_static STATIC ${LIBXMP_SRC_FULL})
+    list(APPEND XMP_INSTALLS xmp_static)
+    set_target_properties(xmp_static PROPERTIES OUTPUT_NAME xmp C_STANDARD 99)
```

Comment:
we do _not_ need c99, as far as I remember

Reply:
I had to set C99 a while ago, because my main build where I used this library as a sub-directory, uses C90, however, libxmp was failed in several cases while I initially wrote the building code. And so, at libxmp, I had to set the C99 explicitly to make it is buildable.

Reply:
That way you are restricting libxmp to c99+, which it does not need.

Reply:
Just now I tried to set the C90 mode, and it got failed, because there is a code that uses C99 features such as `for(int i = 0; ...`

![Uploading Снимок экрана от 2021-07-13 10-58-18.png…]()

Reply:
Then it's your code, or whatever code that uses c99 and it's their fault not requiring c99,
but libxmp does _not_ use c99 features, so do _not_ enforce it here.

Reply:
Sorry, picture wasn't uploaded... That was the `src/lutgen.c` file at libxmp:
![Снимок экрана от 2021-07-13 10-58-18](https://user-images.githubusercontent.com/6751442/125415193-b8240ec9-e75a-4d2c-9fff-30afbf0ba67a.png)

Reply:
This: https://github.com/libxmp/libxmp/blob/master/src/lutgen.c#L275

But wait... it has `main()`, wat the heck I listed it? Gonna check some...

Reply:
OK then, I'll fix lutgen.c to build in c89 mode.

However: lutgen.c is _not_ part of libxmp library build.

Reply:
So, once I removed that lutgen.c from the list, and the library gets built successfully with C90, thanks for pointing this issue, I had to add the jank file (by the way, what it's the purpose here? why it laying here as a trap for noobs?)

Reply:
> However: lutgen.c is not part of libxmp library build.

So, what it's the purpose? Maybe you should move it out of the "src" directory into somewhere also?

Reply:
> So, what it's the purpose? Maybe you should move it out of the "src" directory into somewhere also?

It is used to generate `precomp_lut.h`

Reply:
I see